src/app/OperationalDeviceProxy.cpp

```diff
@@ -309,6 +309,7 @@ void OperationalDeviceProxy::OnSessionEstablished(const SessionHandle & session)
         return; // Got an invalid session, do not change any state
 
     MoveToState(State::SecureConnected);
+    mInitParams.sessionManager->ShiftToSession(session);
```

Comment:
This (and the matching change at `CASEServer`) is shifting sessions at 1 level too high IMHO. Instead, having `SecureSession` call into `SecureSessionTable` (it already has a reference) upon activation, and having that walk the list of sessions and shifting those that match would cover both cases with a single bit of logic I'd think.